consensus/src/simplex/engine.rs

```diff
@@ -89,6 +96,14 @@ impl<
             },
         );
 
+        let observer_config = cfg.listen_addr.map(|listen_addr| observer::Config {
```

Comment:
This should require no changes to `simplex` and can be defined entirely over the `Reporter` stream 👍